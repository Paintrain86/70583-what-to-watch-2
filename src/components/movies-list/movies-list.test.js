import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';

describe(`MoviesList`, () => {
  it(`component is rendered correctly`, () => {
    const testMovies = [
      {
        id: 1,
        title: `Тестовая киношка`,
        poster: `images/image.jpg`,
        previews: [
          {
            src: `movies/movie.mp4`,
            type: `video/mp4`
          }
        ]
      },
      {
        id: 2,
        title: `Тестовая киношка2`,
        poster: `images/image2.jpg`,
        previews: [
          {
            src: `movies/movie2.mp4`,
            type: `video/mp4`
          }
        ]
      }
    ];

    const markup = renderer
      .create(<MoviesList movies={testMovies}/>)
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});
