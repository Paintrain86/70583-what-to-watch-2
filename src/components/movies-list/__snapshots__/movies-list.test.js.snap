// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList MoviesList should render correctly 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card active"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <video
          muted={true}
          poster="images/image.jpg"
          width="100%"
        >
          <source
            src="movies/movie.mp4"
            type="video/mp4"
          />
          Sorry, your browser is a piece of shit
        </video>
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Тестовая киношка
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card "
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
        onClick={[Function]}
      >
        <video
          muted={true}
          poster="images/image2.jpg"
          width="100%"
        >
          <source
            src="movies/movie2.mp4"
            type="video/mp4"
          />
          Sorry, your browser is a piece of shit
        </video>
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
          onClick={[Function]}
        >
          Тестовая киношка2
        </a>
      </h3>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
